pkg/semver/semver.go

```diff
@@ -67,3 +67,33 @@ func (v *Version) SamePrerelease(v2 *Version) bool {
 func (v *Version) Equal(v2 *Version) bool {
 	return v.SamePrerelease(v2) && v.Buildmetadata == v2.Buildmetadata
 }
+
+func (v *Version) GreaterThan(v2 *Version) bool {
+	return v.Compare(v2) == 1
+}
+
+func (v *Version) LessThan(v2 *Version) bool {
+	return v.Compare(v2) == -1
+}
+
+func (v *Version) Compare(v2 *Version) int {
```

Comment:
I like this method
however, I would vote to re-implement it without nested if clauses
I think it would help with readability
I'm ok if you prefer to do it in a separate PR

Reply:
I opted to write it this way to avoid repetitive checks. For example, we would have to check that the major versions are not equal twice to satisfy both the `GreaterThan` and the `LessThan` case if it's broken into separate if statements. If that's preferred for readability, I can update it in my next PR that builds off of this one.